core/src/main/java/org/axonframework/config/SagaConfiguration.java

```diff
@@ -411,6 +448,13 @@ private SagaConfiguration(Class<S> sagaType, ProcessorInfo processorInfo, Class<
     public void initialize(Configuration config) {
         this.config = config;
         eventProcessorRegistry().registerHandlerInvoker(processorInfo.getName(), c -> sagaManager.get());
+        eventProcessorRegistry().registerTokenStore(processorInfo.getName(), c -> tokenStore.get());
+        eventProcessorRegistry().configureMessageMonitor(processorInfo.getName(),
+                                                         c -> (MessageMonitor<Message<?>>) messageMonitor.get());
```

Comment:
Is the cast needed to get the generics right?

Reply:
Si.